src/client/src/registerServiceWorker.js

```diff
@@ -0,0 +1,65 @@
+const isLocalhost = Boolean(
+	window.location.hostname === 'localhost' ||
+    window.location.hostname === '[::1]' ||
+	window.location.hostname
+		.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/)
+)
+
+export default function register() {
+	if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+		const publicUrl = new URL(process.env.PUBLIC_URL, window.location)
+		if (publicUrl.origin !== window.location.origin) return
+
+		window.addEventListener('load', () => {
+			const swUrl = `${process.env.PUBLIC_URL}/service-worker.js`
+
+			if (isLocalhost) {
+				checkValidServiceWorker(swUrl)
+				navigator.serviceWorker.ready.then(() => {})
```

Comment:
`navigator.serviceWorker.ready.then(_.noop)`? Why do we do that at all?

Reply:
I'm unsure how much of this we can remove before things break. Make it a ticket perhaps?